pkg/scheduler/framework/plugins/dynamicresources/extended/cache.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extended
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	klog "k8s.io/klog/v2"
+	fwk "k8s.io/kube-scheduler/framework"
+)
+
+// ExtendedResourceCache maintains a global cache of extended resource to device class mappings
+// that can be accessed by event handlers without requiring a scheduling cycle.
+type ExtendedResourceCache struct {
+	mutex sync.RWMutex
+	// mapping maps extended resource name to device class name
+	mapping    map[v1.ResourceName]string
```

Comment:
JFYI: There is [ongoing debate](https://github.com/kubernetes/kubernetes/pull/133734/files#r2347169756) about changing this mapping to `map[v1.ResourceName]*resourceapi.DeviceClass`.